pkg/tls/certificate.go

```diff
@@ -237,3 +245,30 @@ func (c *Certificates) Set(value string) error {
 func (c *Certificates) Type() string {
 	return "certificates"
 }
+
+// DecodePrivateKey decodes the key if encrypted.
+func DecodePrivateKey(keyContent []byte, passphrase string) ([]byte, error) {
+	var keyDecoded []byte
+	var err error
+	if len(passphrase) == 0 {
+		// No passphrase provided, return provided key.
+		return keyContent, nil
+	}
+
+	keyPemBlock, _ := pem.Decode(keyContent)
+	if x509.IsEncryptedPEMBlock(keyPemBlock) {
+		keyDecoded, err = x509.DecryptPEMBlock(keyPemBlock, []byte(passphrase))
+		if err != nil {
+			return nil, fmt.Errorf("unable to decrypt TLS certificate key : %w", err)
+		}
+
+		keyDecoded = pem.EncodeToMemory(&pem.Block{
+			Type:  keyPemBlock.Type,
+			Bytes: keyDecoded,
+		})
+	} else {
+		keyDecoded = pem.EncodeToMemory(keyPemBlock)
```

Comment:
Hi @dtomcej,
Returning directly the key content must work as well (at least it's not breaking unit tests), that would make the call to EncodeToMemory not necessary, what do you think ?

Reply:
I'm not sure what you mean.

If the key block is not encoded, it is returned after being encoded from a `pem.block` to `[]byte`.

`keyPemBlock` is not a byte array.

Reply:
I was thinking of something like that :
```go
if len(passphrase) == 0 {
    // No passphrase provided, return provided key.
    return keyContent, nil
}

keyPemBlock, _ := pem.Decode(keyContent)
if x509.IsEncryptedPEMBlock(keyPemBlock) {
    keyDecoded, err := x509.DecryptPEMBlock(keyPemBlock, []byte(passphrase))
    if err != nil {
        return nil, fmt.Errorf("unable to decrypt TLS certificate key : %w", err)
    }

    return pem.EncodeToMemory(&pem.Block{
        Type:  keyPemBlock.Type,
        Bytes: keyDecoded,
    }), nil
} 

return keyContent, nil
```

WDYT ?